packages/web-components/src/components/masthead/megamenu-heading.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, customElement, LitElement, property } from 'lit-element';
+import ArrowRight24 from 'carbon-web-components/es/icons/arrow--right/24.js';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener.js';
+import ddsSettings from '../../internal/vendor/@carbon/ibmdotcom-utilities/utilities/settings/settings';
+import styles from './masthead.scss';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * MegaMenu Heading.
+ *
+ * @element dds-megamenu-heading
+ */
+@customElement(`${ddsPrefix}-megamenu-heading`)
+class DDSMegaMenuHeading extends HostListenerMixin(LitElement) {
+  /**
+   * Megamenu heading href.
+   */
+  @property({ reflect: true })
+  href = '';
+
+  /**
+   * Megamenu heading text.
+   */
+  @property({ reflect: true })
+  title = '';
+
+  protected _hasContent = false;
+
+  /**
+   * Arrow icon to use when presented as link.
+   */
+  protected _arrowIcon = ArrowRight24();
+
+  /**
+   * Render heading as link.
+   */
+  protected renderLink() {
+    const { _arrowIcon } = this;
+    return html`
+      <a href="${this.href}" data-attribute1="headerNav" data-attribute2="FlatHdline" data-attribute3="${this.title}">
+        ${this.title} ${_arrowIcon}
+      </a>
+    `;
+  }
+
+  /**
+   * Render heading as plain text.
+   */
+  protected renderPlain() {
+    return html`
+      ${this.title}
+    `;
+  }
+
+  protected _handleSlotChange(event) {
+    this._hasContent = (event.target as HTMLSlotElement).assignedNodes().filter(child => child?.textContent?.trim()).length > 0;
+  }
+
+  render() {
+    const hasHref = Boolean(this.href);
+    return html`
+      <h2 ?data-has-href="${hasHref}">
```

Comment:
Should we add a way to control the heading level dynamically? Right now everything loads in as an H2 which implies they're all equally important in the hierarchy, but the visual styling and layout don't suggest that they should be.
<img width="1288" alt="Screenshot 2022-12-05 at 1 49 04 PM" src="https://user-images.githubusercontent.com/25532785/205719628-93f038bc-1f90-4545-aa0b-f3957c357099.png">

Reply:
Good catch. I added a pretty rudimentary way to control this in [ecf24a3](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/9627/commits/ecf24a38f7c4a60d1c5bb36adc320396356ed3b5).